spring-cloud-gcp-autoconfigure/src/main/java/com/google/cloud/spring/autoconfigure/pubsub/GcpPubSubAutoConfiguration.java

```diff
@@ -131,6 +131,7 @@ public ExecutorProvider publisherExecutorProvider(
 
 	@Bean
 	@ConditionalOnMissingBean(name = "pubsubSubscriberThreadPool")
+	@Deprecated
```

Comment:
Typically we include a javadoc comment pointing to what is replacing it.

Reply:
Hm I think keeping these wouldn't make a difference since we fall back to the default value in `PubSubEventSpecificProperties` if the subscription-specific settings are not set:
```
public Subscriber getSubscriber(String name) {
		if (!this.properties.containsKey(name)) {
			Properties property = new Properties();
			property.setSubscriber(this.subscriber);
			this.properties.put(name, property);
		}

		return this.properties.get(name).getSubscriber();
	}
```

Reply:
Update: After having discussed this offline, we decided to keep the beans but deprecate them and log a message informing the user to use `application.properties` when the bean is set. The subscription-specific properties option will not be available to those who directly use the beans.

Reply:
I think the latest was that we can remove our autoconfigured bean `subscriberExecutorProvider`, and mark the place that uses it as accepting `Optional`.

If we don't remove the bean, it will always get injected into the subscriber factory and override the per-subscription settings, won't it?

Reply:
Ah you're absolutely right! Sorry, I keep mixing up the definition of deprecation vs removal. Yes, we need to remove our bean and mark the place where it is used as Optional.